src/main/java/org/opensearch/knn/plugin/stats/suppliers/KNNClusterLevelCircuitBreakerSupplier.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.stats.suppliers;
+
+import lombok.AllArgsConstructor;
+import org.opensearch.knn.plugin.transport.KNNCircuitBreakerTrippedAction;
+import org.opensearch.knn.plugin.transport.KNNCircuitBreakerTrippedRequest;
+import org.opensearch.knn.plugin.transport.KNNCircuitBreakerTrippedResponse;
+import org.opensearch.transport.client.Client;
+
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+
+@AllArgsConstructor
+public class KNNClusterLevelCircuitBreakerSupplier implements Supplier<Boolean> {
+
+    private final Client client;
+
+    @Override
+    public Boolean get() {
+        try {
+            KNNCircuitBreakerTrippedResponse knnCircuitBreakerTrippedResponse = client.execute(
```

Comment:
TODO: This is going to have to somehow be made to be async, which will complicate things.

Reply:
I made this change. Its a bit messy - will clean up a bit.